tests/streaming/test_grpc_stream_broadcaster.py

```diff
@@ -0,0 +1,198 @@
+
+    def __init__(
+        self, error: Exception, ready_event: asyncio.Event, num_successes: int = 0
+    ):
+        self._error = error
+        self._ready_event = ready_event
+        self._num_successes = num_successes
+        self._current = -1
+
+    async def __anext__(self) -> int:
+        self._current += 1
+        await self._ready_event.wait()
+        if self._current >= self._num_successes:
+            raise self._error
+        return self._current
+
+
+async def test_streaming_success(
+    ok_helper: streaming.GrpcStreamBroadcaster[
+        int, str
+    ],  # pylint: disable=redefined-outer-name
+    no_retry: MagicMock,  # pylint: disable=redefined-outer-name
+    receiver_ready_event: asyncio.Event,  # pylint: disable=redefined-outer-name
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test streaming success."""
+    caplog.set_level(logging.INFO)
+    items: list[str] = []
+    async with asyncio.timeout(1):
+        receiver = ok_helper.new_receiver()
+        receiver_ready_event.set()
+        async for item in receiver:
+            items.append(item)
+    no_retry.next_interval.assert_called_once_with()
+    assert items == [
+        "transformed_0",
+        "transformed_1",
+        "transformed_2",
+        "transformed_3",
+        "transformed_4",
+    ]
+    assert caplog.record_tuples == [
+        (
+            "frequenz.client.base.streaming",
+            logging.ERROR,
+            "test_helper: connection ended, retry limit exceeded (mock progress), "
+            "giving up. Stream exhausted.",
+        )
+    ]
+
+
+class _NamedMagicMock(MagicMock):
+    """Mock with a name."""
+
+    def __str__(self) -> str:
+        return self._mock_name  # type: ignore
+
+    def __repr__(self) -> str:
+        return self._mock_name  # type: ignore
+
+
+@pytest.mark.parametrize("successes", [0, 1, 5])
+@pytest.mark.parametrize(
+    "error_spec",
+    [
+        (
+            grpc.aio.AioRpcError(
+                code=_NamedMagicMock(name="mock grpcio code"),
+                initial_metadata=MagicMock(),
+                trailing_metadata=MagicMock(),
+                details="mock details",
+                debug_error_string="mock debug_error_string",
+            ),
+            "<AioRpcError of RPC that terminated with:\n"
+            "\tstatus = mock grpcio code\n"
+            '\tdetails = "mock details"\n'
+            '\tdebug_error_string = "mock debug_error_string"\n'
+            ">",
+        ),
+        (
+            grpclib.GRPCError(
+                status=_NamedMagicMock(name="mock grpclib status"),
+                message="mock grpclib error",
+                details="mock grpclib details",
+            ),
+            "(mock grpclib status, 'mock grpclib error', 'mock grpclib details')",
+        ),
+    ],
+    ids=["grpcio", "grpclib"],
+)
+async def test_streaming_grpcio_error(  # pylint: disable=too-many-arguments
```

Comment:
Maybe `grpc_error` or just `error` instead?
```suggestion
async def test_streaming_grpc_error(  # pylint: disable=too-many-arguments
```

Reply:
Will change to  `test_streaming_error`.